src/request.rs

```diff
@@ -304,13 +304,53 @@ impl RequestResult {
     ///
```

Comment:
I appreciate this api because it allows us to use a basic command (ensure_status) to perform the meat of most of our needs, and have more specific commands.
Maybe it can mean that ensure_status could be renamed to reflect the 2 things it does, maybe it's not needed, not sure about that.